### PR TITLE
[tests-only]tests: add tests for removing share permission in link share

### PR DESCRIPTION
**Pull request by SwikritiT**
part of: https://github.com/owncloud/ocis/issues/8008

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/deletePermissions.feature [hunk 1]
```diff
@@ -125,3 +125,79 @@ Feature: Remove access to a drive item
...
+    Examples:
+      | role           |
+      | view           |
+      | edit           |
+      | upload         |
+      | createOnly     |
+      | blocksDownload |
+
+
+  Scenario Outline: user removes access to a folder in link share
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/deletePermissions.feature:149]:
```suggestion
  Scenario Outline: user removes access to a file in link share
```

File: tests/acceptance/features/apiSharingNg/deletePermissions.feature [hunk 2]
```diff
@@ -125,3 +125,79 @@ Feature: Remove access to a drive item
...
+    Examples:
+      | role           |
+      | view           |
+      | edit           |
+      | upload         |
+      | createOnly     |
+      | blocksDownload |
+
+
+  Scenario Outline: user removes access to a folder in project space in link share
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSharingNg/deletePermissions.feature:187]:
```suggestion
  Scenario Outline: user removes access to a file in project space in link share
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 3]
```diff
@@ -329,33 +329,36 @@ public function userSetsOrUpdatesFollowingPasswordForLastLinkShareUsingTheGraphA
...
 			? $this->spacesContext->getResourceId($sharer, $space, $resource)
 			: $this->spacesContext->getFileId($sharer, $space, $resource);
-		$permId = $this->featureContext->shareNgGetLastCreatedUserGroupShareID();
-		$this->featureContext->setResponse(
+		$permId = null;
+		if ($shareType === 'link') {
+			$permId = $this->featureContext->shareNgGetLastCreatedLinkShareID();
+		} else {
+			$permId = $this->featureContext->shareNgGetLastCreatedUserGroupShareID();
+		}
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:360]:
```suggestion
		$permId = ($shareType === 'link') ? $this->featureContext->shareNgGetLastCreatedLinkShareID() : $this->featureContext->shareNgGetLastCreatedUserGroupShareID();
		}
```
Or we can do in this way

**Review by grgprarup — APPROVED** [review]
Minor suggestion, other than LGTM :+1:

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 4]
```diff
@@ -364,7 +365,56 @@ public function userRemovesSharePermissionOfUserFromResourceOfSpaceUsingGraphAPI
...
+	 * @param string $sharer
+	 * @param string $resourceType
+	 * @param string $resource
+	 * @param string $space
+	 *
+	 * @return void
+	 * @throws JsonException
+	 * @throws GuzzleException
+	 */
+	public function removeSharePermissionOfAResourceInLinkShareUsingGraphAPI(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:410]:
```suggestion
	public function userRemovesSharePermissionOfAResourceInLinkShareUsingGraphAPI(
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 5]
```diff
@@ -329,33 +329,34 @@ public function userSetsOrUpdatesFollowingPasswordForLastLinkShareUsingTheGraphA
...
+	 * @param string|null $sharee can be both user or group
 	 *
-	 * @return void
+	 * @return ResponseInterface
+	 * @throws GuzzleException
 	 * @throws JsonException
-	 * @throws \GuzzleHttp\Exception\GuzzleException
 	 */
-	public function userRemovesSharePermissionOfUserFromResourceOfSpaceUsingGraphAPI(
+	public function removeSharePermissionOfAResource(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:343]:
```suggestion
	public function removeSharePermission(
```

**Review by saw-jan — APPROVED** [review]

**Pull request merged**